common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -335,6 +349,32 @@ protected void processRedirectUrl(@NonNull final WebView view, @NonNull final St
         //the TokenTask should be processed at after the authorization process in the upper calling layer.
     }
 
+    /**
+     * Launch the browser with the given action URI and code.
+     * <p>
+     * From the query parameters, extract the action URI and code,
+     * The constructs the URI with the action URI and code.
+     *
+     * @param url The URL to be opened in the browser.
+     */
+    private void processSwitchBrowserRequest(@NonNull final String url) {
+        final String methodTag = TAG + ":processSwitchBrowserRequest";
+        try {
+            mSwitchBrowserHandler.processChallenge(
+                    SwitchBrowserChallenge.constructFromRedirectUrl(url)
+            );
+        } catch (final Throwable throwable) {
+            Logger.error(methodTag, "Switch browser challenge could not be processed.", throwable);
+            final String errorCode;
+            if (throwable instanceof IErrorInformation) {
+                errorCode = ((IErrorInformation) throwable).getErrorCode();
+            } else {
+                errorCode = MALFORMED_URL;
```

Comment:
If we ever get here then is it guaranteed to always be a case of malformed url? We are catching generic throwable so I'm not sure if that's always the case

Reply:
changing error code